File: 413-arithmetic-slices/413-arithmetic-slices.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& nums) {
        int n = nums.size();
        int ans = 0;
        vector<int> dp(n);
        for(int i=2;i<n;i++){
            if(nums[i]-nums[i-1] == nums[i-1] - nums[i-2]){
                dp[i] = 1 + dp[i-1];
                ans+=dp[i];
            }
        }
        
        return ans;
    }
};

File: 413-arithmetic-slices/413-arithmetic-slices.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numberOfArithmeticSlices = function(nums) {
    let n = nums.length;
    let ans = 0;
    for(let i=2, prev = nums[1]-nums[0], curCount = 0;i<n;i++){
        let dif = nums[i] - nums[i-1];
        dif === prev ? (ans+=++curCount):(prev = dif,curCount=0);
    }
    return ans;
};